### PR TITLE
Adding pod description section to README.MD

### PR DESCRIPTION
**Pull request by MateusAndrade**
# Overview
Adding a quickly description about using cocoapods with react-native-share since the example project now uses it on ios.

### CODE REVIEW

**Comment by mikehardy** [comment]
There are some typos in there but this is a good start - in my comments on the example commit though, I was thinking the example itself needs a README (`./example/README.md`) that says "To run this on ios, you must have cocoapods installed and run this command `cd ios && pod install` prior to running `react-native run-android`

**Comment by MateusAndrade** [comment]
> There are some typos in there but this is a good start - in my comments on the example commit though, I was thinking the example itself needs a README (`./example/README.md`) that says "To run this on ios, you must have cocoapods installed and run this command `cd ios && pod install` prior to running `react-native run-android`

Sure! I was thinking on writing this too, but i forget

**Comment by MateusAndrade** [comment]
@mikehardy  I added a README.MD and a npm script to install pods on the example.

Also, i was digging on some issues, and tried to run a project using a local version of react-native-share. But i wasn't able to do this. How you manage to make changes on your fork of react-native-share, and on example project use this local version of the rn-share?

**Comment by mikehardy** [comment]
> @mikehardy I added a README.MD and a npm script to install pods on the example.

Awesome, thank you! Those look great

> Also, i was digging on some issues, and tried to run a project using a local version of react-native-share. But i wasn't able to do this. How you manage to make changes on your fork of react-native-share, and on example project use this local version of the rn-share?

Yeah... this is not easy - the metro bundler does not handle symlinks, so you can't symlink from node_modules/react-native-share inside the example back up to the parent. I have tried making the dependency a relative path ("../") but that didn't work.

So what I usually do is manually copy the file I am working on from the module into node_modules/react-native-share inside the example, and reload. I feel a little dirty every time I do it :smile: but it works. I think metro is about to get more power in this area to make this easier, until then if there's another way, I'd love to know it

**Comment by MateusAndrade** [comment]
> So what I usually do is manually copy the file I am working on from the module into node_modules/react-native-share inside the example, and reload. I feel a little dirty every time I do it  but it works. I think metro is about to get more power in this area to make this easier, until then if there's another way, I'd love to know it

I think i found a better way. Just edit your `settings.gradle` and on the project you are making your changes. change the:

```gradle
project(':react-native-share').projectDir = new File(rootProject.projectDir, '../node_modules/your-package-here/android')
```
To

```gradle
project(':react-native-share').projectDir = new File(rootProject.projectDir, ''YOUR_PROJECT_PATH/android")
```

I tested it changing the open method of rn-share adding a simple Log.D and it WORKED! =D

Still a bit ugly...but its more practical. hahahahaha

**Comment by MateusAndrade** [comment]
Im doing some research on a way to doing this on IOS and i found that MAYBE, you can achieve the same easily if you are using cocoapods. Just editing:

```ruby
  pod 'RNShare', :path => '../PROJECT-PATH-NPM'
```

To

```ruby
  pod 'RNShare', :path => '../PROJECT-PATH-LOCAL'
```

The only thing, is by what i see you will need to point to the folder where your podspec is located. Since im a really newbie to XCode, etc i can't confirm this. :/

**Comment by mikehardy** [comment]
I can imagine a lot of hacks around it (the gradle one is clever, nice!) but for me it's usually just up-arrow+return in my shell after I build the copy command, so in practice it works, and then I don't have accidental changes dirtying up my local git workspace

**Comment by MateusAndrade** [comment]
> I can imagine a lot of hacks around it (the gradle one is clever, nice!) but for me it's usually just up-arrow+return in my shell after I build the copy command, so in practice it works, and then I don't have accidental changes dirtying up my local git workspace

I'm a bit traumatized in doing this copy-and-paste process and I'd rather leave everything as possible as possible. But let's see what this update in metro-bundler will improve on this hahaha

**Comment by MateusAndrade** [comment]
Updating the branch with the master. @mikehardy, @jgcmarins anyone can give a look at this to merge on master? I looked on others prs, and merging this looks good to finally release a new version

File: README.md [hunk 1]
```diff
@@ -51,6 +51,27 @@ If you want to share Whatsapp and Mailto, you should write `LSApplicationQueries
...
+  pod 'RNShare', :path => '../node_modules/react-native-share'
+```
+
+After that, just run a `pod install` or `pod udpate` to get up and running with react-native-share. 
+
+Then run a `react-native link react-native-share`, and doing the steps 6 and 7.
+
+You can also see our example to sse how you need to setup your podfile.
+
+Btw, i also recommend reading this (amazing article)[https://shift.infinite.red/beginner-s-guide-to-using-cocoapods-with-react-native-46cb4d372995] about how pods and rn. =D
```
Comment (jgcmarins) [inline README.md]: typo on `i`?
maybe it's better `I` or `We`

Comment (MateusAndrade) [reply README.md]: fixed

File: README.md [hunk 2]
```diff
@@ -51,6 +51,27 @@ If you want to share Whatsapp and Mailto, you should write `LSApplicationQueries
     ```
 8. Run your project (`Cmd+R`)
 
+
+#### iOS Install(using Pods)
+
+If you wish, you can use [cocopoads](https://cocoapods.org/) to use react-native-share.
+
+You just need to add to your Podfile thr react-native-share dependencie.
```
Comment (mikehardy) [inline README.md]: thr -> the + dependencie -> dependency

File: README.md [hunk 3]
```diff
@@ -51,6 +51,27 @@ If you want to share Whatsapp and Mailto, you should write `LSApplicationQueries
...
+```ruby
+  # React-Native-Share pod
+  pod 'RNShare', :path => '../node_modules/react-native-share'
+```
+
+After that, just run a `pod install` or `pod udpate` to get up and running with react-native-share. 
+
+Then run a `react-native link react-native-share`, and doing the steps 6 and 7.
+
+You can also see our example to sse how you need to setup your podfile.
```
Comment (mikehardy) [inline README.md]: sse -> see

File: README.md [hunk 4]
```diff
@@ -51,6 +51,27 @@ If you want to share Whatsapp and Mailto, you should write `LSApplicationQueries
...
+
+Btw, we also recommend reading this (amazing article)[https://shift.infinite.red/beginner-s-guide-to-using-cocoapods-with-react-native-46cb4d372995] about how pods and rn. =D
```
Comment (mikehardy) [inline README.md]: about how pods and rn work together

File: example/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+To get this project running on Android just follow these steps
+
+1. `yarn` or `npm i`
+2. `yarn start` or `npm start`
+3. `react-native run-android`
+
+
+#IOS
+
+Since we are using cocoapods to manage our ios dependencies. After installing it, you will need to reproduce these steps:
```
Comment (mikehardy) [inline example/README.md]: Since we -> We

File: example/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+1. `yarn` or `npm i`
+2. `yarn start` or `npm start`
+3. `cd ios && pod install && cd ..`
+4. `react-native run-ios`
+
+You can also run `yarn install-pods` to install the ios dependencies.
+
+If you have any doubt about installing pods, you call check our README.MD on the repo to get trought this.
```
Comment (mikehardy) [inline example/README.md]: trought -> through

**Comment by MateusAndrade** [comment]
> Please let me be the first to say that I speak English, but only intermediate Spanish. I think English is not your mother tongue, and I very keenly feel how difficult it is to do anything in a non-native language. You should see my Spanish ;-). So grammar issues sure but it would have worked without them. I just saw someone pointed out a small grammar thing and thought, what the heck, let's go all the way :-)

hahahaha no problem. I also must apologize for my bad english, has been sometime since i dont practice. Btw, thanks for pointing the grammar errors.

@mikehardy  you should give a try at Brazilian portuguese anytime on react-brasil-slack hahahaha 😄

**Comment by mikehardy** [comment]
My fintech startup (here in Ecuador) is going to try to move into Brazil as soon as our product is solid enough and we figure out pt-BR ;-). Tenho a certeza que em breve vou falar português terrível.

**Pull request closed without merging**